### PR TITLE
fix(insights): polishing of multiple breakdowns in Trends

### PR DESCRIPTION
**Pull request by skoob13**
## Problem

Fixes the following bugs related to the multiple breakdowns:
- Map view didn't reset
- The number total value viz didn't reset breakdowns
- Smoothing dropdown was always visible for line graphs
- InsightCard had a wrong letter color
- The popup describing session usage didn't apply to multiple breakdowns
- Activity didn't display multiple breakdowns in insights

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Unit and manual testing

### CODE REVIEW

File: frontend/__snapshots__/components-cards-insight-details--trends-world-map--dark.png
Comment (thmsobrmlr) [inline frontend/__snapshots__/components-cards-insight-details--trends-world-map--dark.png:1]: Is this change of colors intentional?

File: frontend/src/lib/components/Cards/InsightCard/InsightDetails.tsx [hunk 1]
```diff
@@ -158,8 +158,8 @@ function SeriesDisplay({
     const hasBreakdown =
         isInsightQueryWithBreakdown(query) &&
         query.breakdownFilter != null &&
-        query.breakdownFilter.breakdown_type != null &&
-        query.breakdownFilter.breakdown != null
+        ((query.breakdownFilter.breakdown_type != null && query.breakdownFilter.breakdown != null) ||
```
Comment (thmsobrmlr) [inline frontend/src/lib/components/Cards/InsightCard/InsightDetails.tsx:161]: Might be good to extract a helper function for `isBreakdown`.

File: frontend/src/lib/components/Cards/InsightCard/InsightDetails.tsx [hunk 2]
```diff
@@ -158,8 +158,8 @@ function SeriesDisplay({
...
+            Array.isArray(query.breakdownFilter.breakdowns))
```
Comment (thmsobrmlr) [inline frontend/src/lib/components/Cards/InsightCard/InsightDetails.tsx:162]: Is it intentional that we check for any array (also empty) here and below we look for arrays with values?

File: frontend/src/scenes/insights/filters/BreakdownFilter/taxonomicBreakdownFilterLogic.ts [hunk 3]
```diff
@@ -234,11 +234,21 @@ export const taxonomicBreakdownFilterLogic = kea<taxonomicBreakdownFilterLogicTy
...
+                    : [newBreakdown]
+
                 props.updateBreakdownFilter({
-                    breakdowns: values.breakdownFilter.breakdowns
-                        ? [...values.breakdownFilter.breakdowns, newBreakdown]
-                        : [newBreakdown],
+                    breakdowns: updatedBreakdowns,
                 })
+
+                if (
```
Comment (thmsobrmlr) [inline frontend/src/scenes/insights/filters/BreakdownFilter/taxonomicBreakdownFilterLogic.ts:245]: Could use a comment explaining why we have this condition here.

**Review by thmsobrmlr — COMMENTED** [review]
Heya @skoob13, thanks for the follow up here! Overall the changes look good. I left a couple of comments inline. In general I think it would be good to dry up the code a bit, as its hard to understand as is. Might also be worth splitting into separate PRs in the future.

File: frontend/__snapshots__/components-cards-insight-details--trends-world-map--dark.png
Comment (skoob13) [reply frontend/__snapshots__/components-cards-insight-details--trends-world-map--dark.png:1]: It's not intentional. I'll check why they've changed.

File: frontend/src/lib/components/Cards/InsightCard/InsightDetails.tsx (hunk 2 above)
Comment (skoob13) [reply frontend/src/lib/components/Cards/InsightCard/InsightDetails.tsx:162]: It is incorrect. Changed to a length condition.

File: frontend/src/lib/components/Cards/InsightCard/InsightDetails.tsx (hunk 1 above)
Comment (skoob13) [reply frontend/src/lib/components/Cards/InsightCard/InsightDetails.tsx:161]: Nice suggestion. I created a type guard for that.

File: frontend/__snapshots__/components-cards-insight-details--trends-world-map--dark.png
Comment (skoob13) [reply frontend/__snapshots__/components-cards-insight-details--trends-world-map--dark.png:1]: The snapshot didn't update after [changing the primary highlight color](https://github.com/PostHog/posthog/pull/23495/) in the app's theme.

**Review by thmsobrmlr — CHANGES_REQUESTED** [review]
Thanks for the follow-up @skoob13 ! This looks good now, with only one change that I think we should make before merging in: The current implementation would break the insight details for the other implementation of multi-breakdowns. I don't think we need to wait for the refactor to the same format and could instead just handle both formats gracefully there.

**Review by thmsobrmlr — APPROVED** [review]
Some CI unhappiness (looks unrelated), otherwise ready to go in 👍

**Pull request merged**